src/OrbitGl/include/OrbitGl/TrackManager.h

```diff
@@ -103,31 +99,39 @@ class TrackManager {
   [[nodiscard]] int FindMovingTrackIndex();
   void UpdateMovingTrackPositionInVisibleTracks();
   void SortTracks();
-  [[nodiscard]] std::vector<ThreadTrack*> GetSortedThreadTracks();
+  [[nodiscard]] std::vector<ThreadTrack*> GetSortedThreadTracks() ABSL_NO_THREAD_SAFETY_ANALYSIS;
+  [[nodiscard]] std::vector<Track*> GetAllTracksNoLock() const ABSL_NO_THREAD_SAFETY_ANALYSIS;
+  ThreadTrack* GetOrCreateThreadTrackNoLock(uint32_t tid) ABSL_NO_THREAD_SAFETY_ANALYSIS;
```

Comment:
I think you don't wanna disable analysis here, but use annotations like `ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_)`. This means this method may only be called when the mutex is locked.

Reply:
Also, if you use these annotations, I would argue you won't need the `NoLock` suffix in the function name, because the compiler will only allow you to call these functions when they have the proper lock acquired.

Reply:
+1 to both comments.

Reply:
Nice! I was sure that I was missing something, I couldn't figure out that ABSL_EXCLUSIVE_LOCKS was the function I was looking for, thanks @beckerhe !! 

About the NoLock comment +1, I would use Internal instead to differentiate with the public function